Testscripts/Windows/PERF-XDP-TX-FORWARD.ps1

```diff
@@ -0,0 +1,152 @@
+
+function Main {
+    try {
+        # PROVISION VMS FOR LISA WILL ENABLE ROOT USER AND WILL MAKE ENABLE PASSWORDLESS AUTHENTICATION ACROSS ALL VMS.
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        Write-LogInfo "Generating constants.sh ..."
+        $constantsFile = "$LogDir\constants.sh"
+
+        foreach ($vmData in $allVMData) {
+            if ($vmData.RoleName -eq "sender") {
+                $masterVM = $vmData
+            }
+
+            Write-LogInfo "VM $($vmData.RoleName) details :"
+            Write-LogInfo "  Public IP : $($vmData.PublicIP)"
+            Write-LogInfo "  SSH Port : $($vmData.SSHPort)"
+            Write-LogInfo "  Internal IP : $($vmData.InternalIP)"
+            Write-LogInfo ""
+
+            Add-Content -Value "$($vmData.RoleName)=$($vmData.InternalIP)" -Path $constantsFile
+            Add-Content -Value "$($vmData.RoleName)SecondIP=$($vmData.SecondInternalIP)" -Path $constantsFile
+        }
+
+        if ($null -eq $masterVM) {
+            Throw "PERF-XDP-TX-FORWARD requires at least one VM with RoleName of sender"
+        }
+
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)
+
+        if ($currentTestData.SetupConfig.Networking -imatch "SRIOV") {
+            $DataPath = "SRIOV"
+        }
+        else {
+            $DataPath = "Synthetic"
+        }
+        # Start XDP Installation
+        $installXDPCommand = @"
+bash ./XDPForwardingTest.sh 2>&1 > ~/xdpConsoleLogs.txt
+. utils.sh
+collect_VM_properties
+"@
+        Set-Content "$LogDir\StartXDPSetup.sh" $installXDPCommand
+        Copy-RemoteFiles -uploadTo $masterVM.PublicIP -port $masterVM.SSHPort `
+            -files "$constantsFile,$LogDir\StartXDPSetup.sh" `
+            -username $user -password $password -upload -runAsSudo
+        # IF Single core then enable single core only
+        $testJob = Run-LinuxCmd -ip $masterVM.PublicIP -port $masterVM.SSHPort `
+            -username $user -password $password -command "bash ./StartXDPSetup.sh" `
+            -RunInBackground -runAsSudo
+        # Terminate process if ran more than 5 mins
+        # TODO: Check max installation time for other distros when added
+        $timer = 0
+        while ($testJob -and ((Get-Job -Id $testJob).State -eq "Running")) {
+            $currentStatus = Run-LinuxCmd -ip $masterVM.PublicIP -port $masterVM.SSHPort `
+                -username $user -password $password -command "tail -2 ~/xdpConsoleLogs.txt | head -1" -runAsSudo
+            Write-LogInfo "Current Test Status: $currentStatus"
+            Wait-Time -seconds 20
+            $timer += 1
+            if ($timer -gt 30) {
+                Throw "XDPFowardingTest did not stop after 10 mins. Please check xdpConsoleLogs."

```

Comment:
if we want to use limit '10' minutes, we should use stopwatch, actually, using current logic in the powershell, it's definitely over 10 minutes.  But just a suggestion only, not a big issue, we can still leave it as it is, if you think it doesn't matter or just no time to change it.

Reply:
It does not have any hard time limit, so I would like to leave it as it is. but thank you for  educating me on this. 😄